lib/service/Service.ts

```diff
@@ -144,26 +144,35 @@ class Service extends EventEmitter {
   }
 
   /**
-   * Get a list of standing peer orders from the order book for a specified trading pair, or for all
-   * trading pairs if no pair is specified.
+   * Get a list of standing peer orders from the order book for a specified trading pair
    */
   public getOrders = ({ pairId, maxResults }: { pairId?: string, maxResults?: number }) => {
-    const ret: OrderArrays = {
-      buyOrders: [],
-      sellOrders: [],
+    checkArgument(pairId !== undefined && pairId !== '', 'pairId has to be specified');
+
+    const result: { [ type: string ]: OrderArrays } = {
+      peerOrders: {
+        buyOrders: [],
+        sellOrders: [],
+      },
+      ownOrders: {
+        buyOrders: [],
+        sellOrders: [],
+      },
     };
-    if (pairId) {
-      const orderArrays = this.orderBook.getPeerOrders(pairId, maxResults ? maxResults : 0);
-      ret.buyOrders.concat(orderArrays.buyOrders);
-      ret.sellOrders.concat(orderArrays.sellOrders);
-    } else {
-      this.orderBook.pairs.forEach((pair) => {
-        const orderArrays = this.orderBook.getPeerOrders(pair.id, maxResults ? maxResults : 0);
-        ret.buyOrders.concat(orderArrays.buyOrders);
-        ret.sellOrders.concat(orderArrays.sellOrders);
-      });
-    }
-    return ret;
+
+    const concatOrders = (orders: OrderArrays, peerOrders: boolean) => {
```

Comment:
Since we're no longer concatenating results from multiple pairs, I don't think we need any of this `concat` stuff any more. You can just do `getPeerOrders` and `getOwnOrders` and return them in an object as `peerOrders` and `ownOrders` properties.